xla/stream_executor/kernel.h

```diff
@@ -586,8 +586,11 @@ std::unique_ptr<KernelArgsPackedArrayBase> PackKernelArgs(
 template <typename ArgType>
 inline absl::StatusOr<std::unique_ptr<KernelArgsPackedArrayBase>>
 PackKernelArgs(absl::Span<const ArgType> args, uint32_t shared_mem_bytes) {
+#if GOOGLE_CUDA && CUDA_VERSION >= 12010
```

Comment:
cuda specific changes should not be in this directory. I checked that this helper function is used only on GPU, maybe it can be moved to the GPU directory, and adjusted separately for ROCM and CUDA? Ideally, we should not use a GOOGLE_CUDA macro.

Reply:
updated